chcemvediet/apps/obligees/tests/test_forms.py

```diff
@@ -1,123 +1,219 @@
+        email = details.find(u'span[@class="chv-obligee-widget-email"]')
+        self.assertIsNone(email.text)
+        no_email = details.find(u'span[@class="chv-obligee-widget-no-email"]')
+        self.assertIsNotNone(no_email)
 
-    @unittest.skip(u'FIXME')
-    def test_new_form_with_custom_widget_class_attributes(self):
+    def test_new_form_with_custom_widget_class_and_attributes(self):
         form = self.FormWithWidgetAttrs()
         rendered = self._render(u'{{ form }}', form=form)
-        self.assertInHTML(u"""
-                <input class="custom-class autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text" custom-attribute="value">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
+        html = lxml.html.fromstring(rendered)
+        widget = html.find(u'.//div'
+                u'[@id="id_obligee"]'
+                u'[@class="chv-obligee-widget custom-class"]'
+                u'[@custom-attribute="value"]')
+        self.assertIsNotNone(widget)
+
+    def test_new_form_with_custom_input_class_and_attributes(self):
+        form = self.FormWithInputAttrs()
+        rendered = self._render(u'{{ form }}', form=form)
+        html = lxml.html.fromstring(rendered)
+        form_control = html.find(u'.//input'
+                u'[@class="custom-class pln-autocomplete form-control"]'
+                u'[@data-autocomplete-url="{url}"]'
+                u'[@name="obligee"]'
+                u'[@data-name="obligee"]'
+                u'[@type="text"]'
+                u'[@value=""]'
+                u'[@custom-attribute="value"]'.format(url=reverse(u'obligees:autocomplete')))
+        self.assertIsNotNone(form_control)
 
-    @unittest.skip(u'FIXME')
     def test_new_form_with_initial_value_as_obligee_instance(self):
         names = [u'aaa', u'bbb', u'ccc', u'ddd']
-        oblgs = [self._create_obligee(name=n) for n in names]
+        oblgs = [self._create_obligee(name=n, street=u'{} street'.format(n), city=u'{} city'.format(n), zip=u'12345', emails=u'{}@a.com'.format(n)) for n in names]
         form = self.Form(initial={u'obligee': oblgs[2]})
         rendered = self._render(u'{{ form }}', form=form)
-        self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text" value="ccc">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
+        html = lxml.html.fromstring(rendered)
+        form_control = html.find(u'.//input'
+                u'[@class="form-control pln-autocomplete"]'
+                u'[@data-autocomplete-url="{url}"]'
+                u'[@name="obligee"]'
+                u'[@data-name="obligee"]'
+                u'[@type="text"]'
+                u'[@value="ccc"]'.format(url=reverse(u'obligees:autocomplete')))
+        self.assertIsNotNone(form_control)
+        details = html.find(u'.//div[@class="chv-obligee-widget-details "]')
```

Comment:
Tu ti ostala medzera na konci class stringu. To je narokom?

Podobne ti taka medzera ostala aj na inych miestach v dalsich testoch.

Reply:
`find` vyzaduje presny obsah attributu, zjavne sa porovnava ako string. Opat riesenim moze byt `find_class` :arrow_up: . Mam to teda prerobit alebo to mozem nechat tak ako to je?
 Ta medzera tam je kvoli tomu, ako je napisany template `obligee_widget_input.html`: `<div class="chv-obligee-widget-details {{ obligee|no:"chv-obligee-widget-hide"}}">`

Reply:
@viliambalaz Chapem. Potom asi naozaj lepsie pouzit `find_class`, pretoze ta medzera je implementacny detail, na ktorom by test nemal zavisiet.